### ISSUE TITLE
Players can't connect with Bedrock without opening another connection first

### ISSUE THREAD

**Issue by Doctacosa** [GeyserMC/Geyser#1036]
**Describe the bug**
Some Bedrock players using Windows 10 Edition aren't able to connect to the server, or even see it as online in their servers list. The only way to make it work is to initiate a connection to the same IP address through any other means, which can be by using a Java client, open a website on the same server, run nmap, or something similar. Once that's done, they have a working window of about 30 minutes to connect. After that duration has elapsed, the server will appear to be inaccessible once again. Two of the affected players seem to have a 100% reproduction rate so far.

Others players don't seem to have this issue. The server is live and active as its playable once the connection is established. No firewall blocking UDP traffic is in place.

Our Geyser installation is on Bungeecord, with Floodgate installed.

**To Reproduce**
1. On Windows 10 Edition, add the server.
2. Try to connect.
3. Fails.
4. Open up the website or connect using the Java client; try again.
5. Works.

**Expected behavior**
Bedrock clients should be able to connect without having to open the website, use the Java client first or any other workaround.

**Server Version**
This server is running Paper version git-Paper-116 (MC: 1.16.1) (Implementing API version 1.16.1-R0.1-SNAPSHOT)

**Geyser Version**
This server is running Geyser version 1.0.0 (git-master-d03f56e) (Java: 1.16.1, Bedrock: 1.16.0)

**Minecraft: Bedrock Edition Version**
v1.16.10

**Additional Context**
https://dump.geysermc.org/n0LaxcsvaIWG2P3XuomOpAS4PjTXe5rI

**Comment by DoctorMacc** [comment]
What error do they get when they try to connect without the workaround?

**Comment by DoctorMacc** [comment]
Also - after this 30 minute window, if they're still connected on the server do they remain on?

**Comment by Doctacosa** [comment]
No special error message, only "Unable to connect to world.", as if the server was offline. Likewise, the "Current Ping" indicator on the servers list is red. Hitting any of the workaround shows the ping time, online players and allows the connection.

Staying online past the 30 minute window works, we're tested that successfully. The access will be denied again only if they disconnect for longer than that duration.

**Comment by DoctorMacc** [comment]
Are you on our Discord? This is definitely not a one-time problem; while our developers have never been able to replicate it we've heard reports of users saying it's a thing.

**Comment by Doctacosa** [comment]
I'm not, but I'm up to join for some back-and-forth if there's somewhere or someone I can contact about this.

I'm lucky enough that two of the affected players are part of the server staff and technically inclined, so they're willing to work through this.

**Comment by DoctorMacc** [comment]
Yes, please join us in our #developers channel. At the very least perhaps one of us can attempt to replicate such a setup.

And Java players don't have this issue, correct?

**Comment by Doctacosa** [comment]
I'll be joining shortly as Doctacosa#8882.

Java players are doing just fine. Geyser has been installed for a few weeks already on a non-standard port while I was running tests, and nobody on the Java side experienced unusual connection problems.

**Comment by DoctorMacc** [comment]
We found this problem also occurred with the vanilla Bedrock server.

"After much prodding, it turns out that my host has an anti-DDoS firewall installed at a lower level than my own rules, which was filtering out UDP traffic. Things like Minecraft Bedrock, who use pure UDP connections, were getting caught in that.

It's designed to only allow traffic if you add a firewall rule to allow it, or if a TCP connection was initiated first (see: opening the website, a control panel, or something similar). This wouldn't have been an issue if Bedrock used TCP to report its online status, as viewing the server entry in the servers list would have been enough to open up the firewall, but here we are.
I tracked down the obscure setting in my host's control panel, added the Bedrock ports to the config, and my test players were able to connect successfully without any workarounds.

I'm with SoYouStart, a OVH subsidiary. The extra firewall settings are exclusive to their GAME line of server hardware."

**Comment by Puremin0rez** [comment]
I'm glad I came across this issue - I was experiencing the same thing. I also use SoYouStart.

To add additional information for anyone who comes across this - the setting is quite hidden so I had to discover it myself based on the above information and thought I'd share its location easier.

1) Sign into your SYS control panel
2) Click "IP" on the top header above your server info (it's the first option in the navigation bar)
3) Click the little gear icon (to the very far right) next to the IPv4 address on the list of IPs
4) Select "Game Mitigation"
5) Add a rule for "minecraftPocketEdition" with a port of "19132"

That solved my issue. What a frustrating issue to diagnose :P
